storage/filesystem.go

```diff
@@ -65,30 +70,44 @@ type Filesystem struct {
 // Ensure the Filesystem implementation is fully featured
 var _ Interface = &Filesystem{}
 
-func NewFilesystem(log logr.Logger, baseDir string) (*Filesystem, error) {
-	f := &Filesystem{
-		log:     log,
-		baseDir: baseDir,
-		// Use the rootfs as the DirFS so that paths passed to both read &
-		// write methods on this struct use a consistent root.
-		fs: os.DirFS("/").(fs.StatFS),
+func NewFilesystem(log logr.Logger, basePath string) (*Filesystem, error) {
+	inmemfsPath := filepath.Join(basePath, "inmemfs")
+
+	f, err := NewFilesystemOnDisk(log, inmemfsPath)
+	if err != nil {
+		return nil, err
 	}
 
-	isMnt, err := mount.New("").IsMountPoint(f.tempfsPath())
+	isMnt, err := mount.New("").IsMountPoint(inmemfsPath)
 	if err != nil {
-		if !os.IsNotExist(err) {
-			return nil, err
-		}
-		if err := os.MkdirAll(f.tempfsPath(), 0700); err != nil {
-			return nil, err
-		}
+		return nil, err
 	}
 
 	if !isMnt {
-		if err := mount.New("").Mount("tmpfs", f.tempfsPath(), "tmpfs", []string{}); err != nil {
+		if err := mount.New("").Mount("tmpfs", inmemfsPath, "tmpfs", []string{}); err != nil {
 			return nil, fmt.Errorf("mounting tmpfs: %w", err)
 		}
-		log.Info("Mounted new tmpfs", "path", f.tempfsPath())
+		log.Info("Mounted new tmpfs", "path", inmemfsPath)
+	}
+
+	return f, nil
+}
+
+// WARNING: should be used only for testing purposes only, as we don't want to store
+// any sensitive data on disk in production.
+func NewFilesystemOnDisk(log logr.Logger, basePath string) (*Filesystem, error) {
+	if !filepath.IsAbs(basePath) {
+		return nil, fmt.Errorf("baseDir must be an absolute path")
```

Comment:
Starting from this PR, we no longer support the weird relative path notation. All paths have to be absolute.

Reply:
see https://github.com/cert-manager/csi-lib/pull/71/files#diff-ebf7d236278a68704d3f90c6e44983c3c663d2beb9ce124a1658cb93ffaa6276R89